### PR TITLE
Convert Dependencies to Gradle Version Catalogs

### LINKED ISSUE #360: Change Gradle Dependency Central Versioning Scheme to Abide by Dependabot Rules
See this: https://github.com/dependabot/dependabot-core/issues/3121

Currently, dependabot can only detect new versions if the dependency is hardcoded with the dependency.

I created an `ext.versions` property to ensure a centralized way of updating versions, but dependabot doesn't deal with this.

Gradle has a new way of declaring central dependencies: https://docs.gradle.org/nightly/userguide/platforms.html#sub:central-declaration-of-dependencies

Not sure if the above info works on our current gradle version...

However, we will probably want to investigate the dependabot way.

### PR DESCRIPTION
**Pull request by epicstar**
Closes #360 

This is the gradle and dependabot official way to solve #311.

Gives us the ability for depend

Lose ability for IDE to check matching Kotlin version. However... I'd rather trade dependabot updates for matching Kotlin version. Easy to check anyway.

### CODE REVIEW

**Comment by epicstar** [comment]
Note: this was quite a large plumbing task but this was necessary to find blockers for a future release.

**Comment by epicstar** [comment]
We can't update retrofit to any version over 2.7.2 because of this issue: https://github.com/square/retrofit/issues/3341

**Comment by epicstar** [comment]
Jake Wharton has recommended to keep OkHttp to the latest 4.x.x version since it's binary compatible with OkHttp 3.x.x which Retrofit 2.7.2 consumes: https://twitter.com/JakeWharton/status/1232161290610728960

If we're wondering why Retrofit is stuck on 2.9.2 for a year or so, it's because Square is currently rewriting Retrofit to kotlin lol.

File: pat-static/src/main/java/com/rectanglel/patstatic/model/PatApiServiceImpl.kt [hunk 1]
```diff
@@ -97,7 +97,7 @@ class PatApiServiceImpl(apiKey: String,
             return { predictionResponse ->
                 predictionResponse
                         .map { it.bustimeResponse }
-                        .map { it.prd }
+                        .map { it!!.prd } // TODO: FIX
```
Comment (mikeantonacci) [inline pat-static/src/main/java/com/rectanglel/patstatic/model/PatApiServiceImpl.kt:100]: 😔

File: pat-static/src/main/kotlin/com/rectanglel/patstatic/patterns/PatternDataManager.kt [hunk 2]
```diff
@@ -83,7 +83,7 @@ class PatternDataManager(dataDirectory: File,
                 .map(PatternResponse::patternResponse)
                 .map { bustimePatternResponse ->
                     try {
-                        val patterns = bustimePatternResponse.ptr
+                        val patterns = bustimePatternResponse!!.ptr // TODO: FIX
```
Comment (mikeantonacci) [inline pat-static/src/main/kotlin/com/rectanglel/patstatic/patterns/PatternDataManager.kt:86]: 😔

**Review by mikeantonacci — APPROVED** [review]
Looks good, let's just make sure to revisit those instances of !!

**Pull request merged**